lib/thrift/binary/framed/client.ex

```diff
@@ -372,4 +401,34 @@ defmodule Thrift.Binary.Framed.Client do
         {:stop, error, s}
     end
   end
+
+  defp maybe_resend_data({:ok, s}, {:reconnect, :call, msg, from}) do
+    case handle_call(msg, from, s) do
+      {:reply, reply, s} ->
+        GenServer.reply(from, reply)
+        {:ok, s}
+
+      {:disconnect, info, error, s} ->
+        GenServer.reply(from, error)
+        disconnect(info, s)
+
+      _ ->
```

Comment:
The handle call could return `{:connect, _, _, _}` here, and we seem to do the wrong thing by returning an ok socket.

For example it could be possible due to a server side bug that the server always closes the connection without sending a response for certain requests - and we would always be seeing it.

Reply:
Actually this is a return from `connect` callback - not from `handle_call`. The result of call to `disconnect(info, s)` returned from `connect` callback [here](https://github.com/cybernetlab/elixir-thrift/blob/allow-reconnect/lib/thrift/binary/framed/client.ex#L158) and it can be only `{:ok, _}` or `{:stop, _}` (or `{:backoff, _}` which is not used)

Reply:
The result is used as the return to `connect` but its calling the `handle_call` callback and matching on its result, which was able to return the connect tuple before. In that situation we would return `{:ok, s}` with wrong socket in the state. I'm not sure if its safe here or in future can to avoid matching on the full result so we can pass state through. We should always match on all results to be sure state isnt messed up.